buildingmotif/database/table_connection.py

```diff
@@ -390,56 +401,88 @@ def add_template_dependency(
         :type args: Dict[str, str]
         :raises ValueError: if all dependee required_params not in args
         :raises ValueError: if dependant and dependency template don't share a
-            library
         """
         self.logger.debug(
             f"Creating depencency from templates with ids: '{template_id}' and: '{dependency_id}'"
         )
         templ = self.get_db_template_by_id(template_id)
-        graph = self.bm.graph_connection.get_graph(templ.body_id)
-        params = get_parameters(graph)
-        dep = self.get_db_template_by_id(dependency_id)
+        if "name" not in args.keys():
+            raise ValueError(
+                f"The name parameter is required for the dependency '{templ.name}'."
+            )
+        # In the past we had a check here to make sure the two templates were in the same library.
+        # This has been removed because it wasn't actually necessary, but we may add it back in
+        # in the future.
+        relationship = DepsAssociation(
+            dependant_id=template_id,
+            dependee_id=dependency_id,
+            args=args,
+        )
+
+        self.bm.session.add(relationship)
+        self.bm.session.flush()
+
+    def check_template_dependencies(self):
+        """
+        Verifies that all template dependencies have valid references to the parameters
+        in the dependency or (recursively) its dependencies. Raises an exception if any
+        issues are found. Checks all templates in the database.
+        """
+        # TODO: maybe optimize this to only check recently added templates
+        # (i.e. maybe those added since last commit?)
+        for db_templ in self.get_all_db_templates():
+            for dep in self.get_db_template_dependencies(db_templ.id):
+                self.check_template_dependency(dep)
+
+    @lru_cache(maxsize=128)
+    def check_template_dependency(self, dep: DepsAssociation):
+        """Create dependency between two templates.
```

Comment:
Should this be "check dependency between two templates"?

Reply:
`check_template_dependency_relationship`?

Reply:
The doc string says "create"

Reply:
Fixed both of these. Thanks for the catch!